gips/data/core.py

```diff
@@ -561,20 +572,33 @@ def _archivefile(cls, filename, update=False):
             newfilename = os.path.join(tpath, bname)
             if not os.path.exists(newfilename):
                 # check if another asset exists
+                # QUESTION: can it be that len(existing) > 1?
+                # IF NOT, then seems like some of these for-loops could go, and
+                # the overall logic could be simplified.
                 existing = cls.discover(asset.tile, d, asset.asset)
                 if len(existing) > 0 and (not update or not existing[0].updated(asset)):
-                    # gatekeeper case:  No action taken because existing assets are in the way
+                    # gatekeeper case:  No action taken because other assets exist
                     VerboseOut('%s: other version(s) already exists:' % bname, 1)
                     for ef in existing:
                         VerboseOut('\t%s' % os.path.basename(ef.filename), 1)
                     otherversions = True
                 elif len(existing) > 0 and update:
-                    # update case:  Remove existing outdated assets and install the new one
+                    # update case:  Remove existing outdated assets
+                    #               and install the new one
                     VerboseOut('%s: removing other version(s):' % bname, 1)
                     for ef in existing:
-                        assert ef.updated(asset), 'Asset is not updated version'
+                        if not ef.updated(asset):
+                            utils.verbose_out(
+                                'Asset {} is not updated version of {}.'
+                                .format(ef.filename, asset.filename) +
+                                ' Remove existing asset to replace.', 2
+                            )
+                            # NOTE: This return makes sense iff len(existing)
+                            # cannot be greater than 1
```

Comment:
Optional:  Maybe put an assertion or warning just before the loop, to confirm that `len(existing)` has a sane value?

Reply:
This note was meant more as something to key in on iff strange behavior occurs.  I guess the assert would make that even more plain and get to the root of things quicker.